Bug: `build:transform` fails when trying to find tsconfig astro/tsconfigs/strict

I'm running rolldown-vite with my astro site -- version [rolldown-vite@7.1.1](https://www.npmjs.com/package/rolldown-vite/v/7.1.1) works fine, and version [rolldown-vite@7.1.2](https://www.npmjs.com/package/rolldown-vite/v/7.1.2) fails with this error:

```
07:50:11 [ERROR] [vite] ✗ Build failed in 1.08s
[UNHANDLEABLE_ERROR] Error: Something went wrong inside native plugin `builtin:transform`. Please report this problem at https://github.com/rolldown/rolldown/issues.
Tsconfig not found astro/tsconfigs/strict

  Location:
    /Users/calebjasik-personal/Git/jasik-2.0/node_modules/.pnpm/rolldown@1.0.0-beta.32/node_modules/rolldown/dist/shared/src-p50l1OMU.mjs:2270:18
  Stack trace:
    at normalizeErrors (file:///Users/calebjasik-personal/Git/jasik-2.0/node_modules/.pnpm/rolldown@1.0.0-beta.32/node_modules/rolldown/dist/shared/src-p50l1OMU.mjs:2270:18)
    at transformToRollupOutput (file:///Users/calebjasik-personal/Git/jasik-2.0/node_modules/.pnpm/rolldown@1.0.0-beta.32/node_modules/rolldown/dist/shared/src-p50l1OMU.mjs:3000:2)
    at async buildEnvironment (file:///Users/calebjasik-personal/Git/jasik-2.0/node_modules/.pnpm/rolldown-vite@7.1.2_@types+node@22.16.4_esbuild@0.25.6_jiti@2.4.2_yaml@2.8.0/node_modules/rolldown-vite/dist/node/chunks/dep-uovcZwnY.js:33495:64)
    at async ssrBuild (file:///Users/calebjasik-personal/Git/jasik-2.0/node_modules/.pnpm/astro@5.12.2_@netlify+blobs@10.0.6_@types+node@22.16.4_idb-keyval@6.2.1_jiti@2.4.2_roll_6a6df5fde7bb99bc7683af133c04e651/node_modules/astro/dist/core/build/static-build.js:181:10)
    at async AstroBuilder.build (file:///Users/calebjasik-personal/Git/jasik-2.0/node_modules/.pnpm/astro@5.12.2_@netlify+blobs@10.0.6_@types+node@22.16.4_idb-keyval@6.2.1_jiti@2.4.2_roll_6a6df5fde7bb99bc7683af133c04e651/node_modules/astro/dist/core/build/index.js:149:48)
 ELIFECYCLE  Command failed with exit code 1.
```

<details>
<summary>Full Build Log</summary>

```
❯ pnpm i && pnpm build
 WARN  1 deprecated subdependencies found: node-domexception@1.0.0
Packages: +5
+++++
Progress: resolved 1660, reused 1500, downloaded 1, added 5, done
 WARN  Issues with peer dependencies found
.
├─┬ astro 5.12.2
│ ├─┬ vitefu 1.0.6
│ │ └── ✕ unmet peer vite@"^3.0.0 || ^4.0.0 || ^5.0.0 || ^6.0.0": found 7.1.2
│ └─┬ unstorage 1.15.0
│   └── ✕ unmet peer @netlify/blobs@"^6.5.0 || ^7.0.0 || ^8.1.0": found 10.0.6
├─┬ @astrojs/react 4.3.0
│ └─┬ @vitejs/plugin-react 4.4.1
│   └── ✕ unmet peer vite@"^4.2.0 || ^5.0.0 || ^6.0.0": found 7.1.2
├─┬ @tailwindcss/vite 4.1.8
│ └── ✕ unmet peer vite@"^5.2.0 || ^6": found 7.1.2
├─┬ astro-embed 0.9.0
│ └─┬ @astro-community/astro-embed-integration 0.8.0
│   └─┬ astro-auto-import 0.4.2
│     └── ✕ unmet peer astro@"^2.0.0 || ^3.0.0-beta || ^4.0.0-beta": found 5.12.2
└─┬ expressive-code-color-chips 0.1.2
  └── ✕ unmet peer @expressive-code/core@^0.37.1: found 0.41.2
Done in 2.3s using pnpm v10.13.1

> jasik-2.0@0.0.1 build /Users/calebjasik-personal/Git/jasik-2.0
> astro build

07:50:09 [@astrojs/netlify] Enabling sessions with Netlify Blobs
[Sentry] You passed in additional options (dsn) to the Sentry integration. This is deprecated and will stop working in a future version. Instead, configure the Sentry SDK in your `sentry.client.config.(js|ts)` or `sentry.server.config.(js|ts)` files.
[Sentry] Enabled source map generation in the build options with `vite.build.sourcemap: 'hidden'`. The source maps will be deleted after they were uploaded to Sentry.
[Sentry] Automatically setting `sourceMapsUploadOptions.filesToDeleteAfterUpload: ["./dist/**/client/**/*.map","./dist/**/server/**/*.map"]` to delete generated source maps after they were uploaded to Sentry.
07:50:09 [WARN] [vite] You or a plugin you are using have set `optimizeDeps.esbuildOptions` but this option is now deprecated. Vite now uses Rolldown to optimize the dependencies. Please use `optimizeDeps.rollupOptions` instead.
[sentry-vite-plugin] Info: Sending telemetry data on issues and performance to Sentry. To disable telemetry, set `options.telemetry` to `false`.
07:50:10 [vite] Re-optimizing dependencies because lockfile has changed
07:50:10 [content] Syncing content
07:50:10 [content] Synced content
07:50:10 [types] Generated 533ms
07:50:10 [build] output: "static"
07:50:10 [build] mode: "server"
07:50:10 [build] directory: /Users/calebjasik-personal/Git/jasik-2.0/dist/
07:50:10 [build] adapter: @astrojs/netlify
07:50:10 [build] Collecting build info...
07:50:10 [build] ✓ Completed in 946ms.
07:50:10 [build] Building server entrypoints...
07:50:10 [WARN] [vite] You or a plugin you are using have set `optimizeDeps.esbuildOptions` but this option is now deprecated. Vite now uses Rolldown to optimize the dependencies. Please use `optimizeDeps.rollupOptions` instead.
[sentry-vite-plugin] Info: Sending telemetry data on issues and performance to Sentry. To disable telemetry, set `options.telemetry` to `false`.
`transformWithEsbuild` is deprecated and will be removed in the future. Please migrate to `transformWithOxc`.
07:50:10 [WARN] [vite] [vite:css][lightningcss] Unknown at rule: @theme
1  |  
2  |        @layer theme, base, components, utilities;
   |       ^
3  |        @import "tailwindcss/theme" layer(theme);
4  |        @import "tailwindcss/utilities" layer(utilities);
07:50:10 [WARN] [vite] [vite:css][lightningcss] Unknown at rule: @theme

07:50:10 [WARN] [vite] [vite:css][lightningcss] Unknown at rule: @tailwind
1  |  
2  |        @layer theme, base, components, utilities;
   |          ^
3  |        @import "tailwindcss/theme" layer(theme);
4  |        @import "tailwindcss/utilities" layer(utilities);
07:50:11 [ERROR] [vite] ✗ Build failed in 1.08s
[UNHANDLEABLE_ERROR] Error: Something went wrong inside native plugin `builtin:transform`. Please report this problem at https://github.com/rolldown/rolldown/issues.
Tsconfig not found astro/tsconfigs/strict

  Location:
    /Users/calebjasik-personal/Git/jasik-2.0/node_modules/.pnpm/rolldown@1.0.0-beta.32/node_modules/rolldown/dist/shared/src-p50l1OMU.mjs:2270:18
  Stack trace:
    at normalizeErrors (file:///Users/calebjasik-personal/Git/jasik-2.0/node_modules/.pnpm/rolldown@1.0.0-beta.32/node_modules/rolldown/dist/shared/src-p50l1OMU.mjs:2270:18)
    at transformToRollupOutput (file:///Users/calebjasik-personal/Git/jasik-2.0/node_modules/.pnpm/rolldown@1.0.0-beta.32/node_modules/rolldown/dist/shared/src-p50l1OMU.mjs:3000:2)
    at async buildEnvironment (file:///Users/calebjasik-personal/Git/jasik-2.0/node_modules/.pnpm/rolldown-vite@7.1.2_@types+node@22.16.4_esbuild@0.25.6_jiti@2.4.2_yaml@2.8.0/node_modules/rolldown-vite/dist/node/chunks/dep-uovcZwnY.js:33495:64)
    at async ssrBuild (file:///Users/calebjasik-personal/Git/jasik-2.0/node_modules/.pnpm/astro@5.12.2_@netlify+blobs@10.0.6_@types+node@22.16.4_idb-keyval@6.2.1_jiti@2.4.2_roll_6a6df5fde7bb99bc7683af133c04e651/node_modules/astro/dist/core/build/static-build.js:181:10)
    at async AstroBuilder.build (file:///Users/calebjasik-personal/Git/jasik-2.0/node_modules/.pnpm/astro@5.12.2_@netlify+blobs@10.0.6_@types+node@22.16.4_idb-keyval@6.2.1_jiti@2.4.2_roll_6a6df5fde7bb99bc7683af133c04e651/node_modules/astro/dist/core/build/index.js:149:48)
 ELIFECYCLE  Command failed with exit code 1.
```

</details>



My package.json looks like:

```json
{
  "name": "jasik-2.0",
  "type": "module",
  "version": "0.0.1",
  "scripts": {
    "dev": "SHOULD_BUILD_DRAFT_AND_FUTURE=true astro dev",
    "start": "SHOULD_BUILD_DRAFT_AND_FUTURE=true astro dev",
    "build": "astro build",
    "preview": "astro preview",
    "astro": "astro",
    "format": "prettier --write .",
    "format:check": "prettier --check .",
    "lint": "pnpm run \"/^lint:.*/\"",
    "lint:js": "eslint .",
    "lint:css": "stylelint --cache --config ./stylelint.config.js '**/*.{css,astro}'",
    "check": "pnpm run \"/^check:.*/\"",
    "check:ts": "tsc -p ./tsconfig.json",
    "check:astro": "astro check -p ./tsconfig.json"
  },
  "engines": {
    "node": ">=18",
    "pnpm": ">=8"
  },
  "dependencies": {
    "@astrojs/check": "0.9.4",
    "@astrojs/markdoc": "0.15.1",
    "@astrojs/netlify": "6.5.2",
    "@astrojs/react": "4.3.0",
    "@astrojs/rss": "4.0.12",
    "@astrojs/sitemap": "3.4.1",
    "@keystatic/astro": "5.0.6",
    "@keystatic/core": "0.5.47",
    "@sentry/astro": "9.39.0",
    "@tailwindcss/vite": "4.1.8",
    "@types/react": "19.1.8",
    "@types/react-dom": "19.1.6",
    "@zachleat/pagefind-search": "github:zachleat/pagefind-search",
    "assert-ts": "0.3.4",
    "astro": "5.12.2",
    "astro-capo": "0.0.1",
    "astro-embed": "0.9.0",
    "astro-expressive-code": "0.41.2",
    "astro-icon": "1.1.5",
    "astro-og-canvas": "0.7.0",
    "astro-pagefind": "1.8.3",
    "canvaskit-wasm": "0.40.0",
    "date-fns": "4.1.0",
    "expressive-code-color-chips": "0.1.2",
    "lightningcss": "1.30.1",
    "lucide-react": "0.522.0",
    "pagefind": "1.3.0",
    "react": "19.1.0",
    "react-dom": "19.1.0",
    "sharp": "0.34.2",
    "shiki": "3.7.0",
    "stylelint": "16.22.0",
    "stylelint-gamut": "1.3.4",
    "tailwindcss": "4.1.8",
    "tiny-decode": "0.1.3",
    "ultrahtml": "1.6.0"
  },
  "packageManager": "pnpm@10.13.1",
  "devDependencies": {
    "@eslint/js": "9.31.0",
    "@ianvs/prettier-plugin-sort-imports": "4.4.2",
    "@iconify-json/ri": "1.2.5",
    "@types/node": "22.16.4",
    "@typescript-eslint/parser": "8.38.0",
    "eslint": "9.31.0",
    "eslint-plugin-astro": "1.3.1",
    "eslint-plugin-jsx-a11y": "6.10.2",
    "postcss-html": "1.8.0",
    "prettier": "3.6.2",
    "prettier-plugin-astro": "0.14.1",
    "prettier-plugin-tailwindcss": "0.6.13",
    "prettier-plugin-toml": "2.0.5",
    "stylelint-config-astro": "1.0.4",
    "stylelint-config-recommended": "16.0.0",
    "typescript": "5.8.3",
    "typescript-eslint": "8.38.0"
  },
  "imports": {
    "#assets/*": [
      "./src/assets/*",
      "./src/assets/*.ts",
      "./src/assets/*.tsx"
    ],
    "#components/*": [
      "./src/components/*",
      "./src/components/*.ts",
      "./src/components/*.tsx"
    ],
    "#utils/*": [
      "./src/utils/*",
      "./src/utils/*.ts",
      "./src/utils/*.tsx"
    ]
  },
  "pnpm": {
    "onlyBuiltDependencies": [
      "@sentry/cli",
      "@tailwindcss/oxide",
      "esbuild",
      "optipng-bin",
      "sharp"
    ],
    "overrides": {
      "vite": "npm:rolldown-vite@7.1.2"
    }
  }
}
```

With rolldown-vite@7.1.9:

```console
❯ pnpm build

> jasik-2.0@0.0.1 build /Users/calebjasik/Git/jasik-2.0
> astro build

11:27:39 [@astrojs/netlify] Enabling sessions with Netlify Blobs
[Sentry] Enabled source map generation in the build options with `vite.build.sourcemap: 'hidden'`. The source maps will be deleted after they were uploaded to Sentry.
[Sentry] Automatically setting `sourceMapsUploadOptions.filesToDeleteAfterUpload: ["./dist/**/client/**/*.map","./dist/**/server/**/*.map"]` to delete generated source maps after they were uploaded to Sentry.
[sentry-vite-plugin] Warning: No auth token provided. Will not create release. Please set the `authToken` option. You can find information on how to generate a Sentry auth token here: https://docs.sentry.io/api/auth/
[sentry-vite-plugin] Warning: No auth token provided. Will not upload source maps. Please set the `authToken` option. You can find information on how to generate a Sentry auth token here: https://docs.sentry.io/api/auth/
11:27:39 [WARN] [vite] [vite:react-babel] We recommend switching to `@vitejs/plugin-react-oxc` for improved performance. More information at https://vite.dev/rolldown
[sentry-vite-plugin] Info: Sending telemetry data on issues and performance to Sentry. To disable telemetry, set `options.telemetry` to `false`.
11:27:40 [vite] Re-optimizing dependencies because vite config has changed
11:27:40 [content] Syncing content
11:27:40 [content] Synced content
11:27:40 [types] Generated 498ms
11:27:40 [build] output: "static"
11:27:40 [build] mode: "server"
11:27:40 [build] directory: /Users/calebjasik/Git/jasik-2.0/dist/
11:27:40 [build] adapter: @astrojs/netlify
11:27:40 [build] Collecting build info...
11:27:40 [build] ✓ Completed in 884ms.
11:27:40 [build] Building server entrypoints...
11:27:40 [WARN] [vite] [vite:react-babel] We recommend switching to `@vitejs/plugin-react-oxc` for improved performance. More information at https://vite.dev/rolldown
[sentry-vite-plugin] Info: Sending telemetry data on issues and performance to Sentry. To disable telemetry, set `options.telemetry` to `false`.
`transformWithEsbuild` is deprecated and will be removed in the future. Please migrate to `transformWithOxc`.
11:27:40 [WARN] [vite] [vite:css][lightningcss] Unknown at rule: @tailwind
1  |  
2  |        @layer theme, base, components, utilities;
   |          ^
3  |        @import "tailwindcss/theme" layer(theme);
4  |        @import "tailwindcss/utilities" layer(utilities);
11:27:40 [WARN] [vite] [vite:css][lightningcss] Unknown at rule: @theme
1  |  
2  |        @layer theme, base, components, utilities;
   |       ^
3  |        @import "tailwindcss/theme" layer(theme);
4  |        @import "tailwindcss/utilities" layer(utilities);
11:27:40 [WARN] [vite] [vite:css][lightningcss] Unknown at rule: @theme

11:27:41 [ERROR] [vite] ✗ Build failed in 1.30s
[UNHANDLEABLE_ERROR] Error: Something went wrong inside native plugin `builtin:transform`. Please report this problem at https://github.com/rolldown/rolldown/issues.
Tsconfig not found astro/tsconfigs/strict

  Location:
    /Users/calebjasik/Git/jasik-2.0/node_modules/.pnpm/rolldown@1.0.0-beta.37/node_modules/rolldown/dist/shared/src-BtNlw_84.mjs:2319:18
  Stack trace:
    at normalizeErrors (file:///Users/calebjasik/Git/jasik-2.0/node_modules/.pnpm/rolldown@1.0.0-beta.37/node_modules/rolldown/dist/shared/src-BtNlw_84.mjs:2319:18)
    at transformToRollupOutput (file:///Users/calebjasik/Git/jasik-2.0/node_modules/.pnpm/rolldown@1.0.0-beta.37/node_modules/rolldown/dist/shared/src-BtNlw_84.mjs:3057:2)
    at async buildEnvironment (file:///Users/calebjasik/Git/jasik-2.0/node_modules/.pnpm/rolldown-vite@7.1.9_@types+node@22.16.4_esbuild@0.25.9_jiti@2.4.2_yaml@2.8.0/node_modules/rolldown-vite/dist/node/chunks/dep-HNGlhFR2.js:33174:64)
    at async ssrBuild (file:///Users/calebjasik/Git/jasik-2.0/node_modules/.pnpm/astro@5.12.2_@netlify+blobs@10.0.10_@types+node@22.16.4_idb-keyval@6.2.1_jiti@2.4.2_rol_a2c32aaf5948e7ea7c89cb0b2a0e6bb7/node_modules/astro/dist/core/build/static-build.js:181:10)
    at async AstroBuilder.build (file:///Users/calebjasik/Git/jasik-2.0/node_modules/.pnpm/astro@5.12.2_@netlify+blobs@10.0.10_@types+node@22.16.4_idb-keyval@6.2.1_jiti@2.4.2_rol_a2c32aaf5948e7ea7c89cb0b2a0e6bb7/node_modules/astro/dist/core/build/index.js:149:48)
 ELIFECYCLE  Command failed with exit code 1.
```

If I disable the native plugins, it works just fine.

Sorry, my mistake — you should set it like this:

```json
"pnpm": {
  "overrides": {
    "vite": "npm:rolldown-vite@7.1.9",
    "rolldown-vite>rolldown": "1.0.0-beta.38"
  }
}
```

here's the console output if you want to verify they're the same

```console
❯ pnpm build

> jasik-2.0@0.0.1 build /Users/calebjasik/Git/jasik-2.0
> astro build

13:25:16 [@astrojs/netlify] Enabling sessions with Netlify Blobs
[Sentry] Enabled source map generation in the build options with `vite.build.sourcemap: 'hidden'`. The source maps will be deleted after they were uploaded to Sentry.
[Sentry] Automatically setting `sourceMapsUploadOptions.filesToDeleteAfterUpload: ["./dist/**/client/**/*.map","./dist/**/server/**/*.map"]` to delete generated source maps after they were uploaded to Sentry.
[sentry-vite-plugin] Warning: No auth token provided. Will not create release. Please set the `authToken` option. You can find information on how to generate a Sentry auth token here: https://docs.sentry.io/api/auth/
[sentry-vite-plugin] Warning: No auth token provided. Will not upload source maps. Please set the `authToken` option. You can find information on how to generate a Sentry auth token here: https://docs.sentry.io/api/auth/
13:25:17 [WARN] [vite] [vite:react-babel] We recommend switching to `@vitejs/plugin-react-oxc` for improved performance. More information at https://vite.dev/rolldown
[sentry-vite-plugin] Info: Sending telemetry data on issues and performance to Sentry. To disable telemetry, set `options.telemetry` to `false`.
13:25:17 [vite] Re-optimizing dependencies because lockfile has changed
13:25:17 [content] Syncing content
13:25:17 [content] Astro config changed
13:25:17 [content] Clearing content store
13:25:17 [content] Synced content
13:25:17 [types] Generated 566ms
13:25:17 [build] output: "static"
13:25:17 [build] mode: "server"
13:25:17 [build] directory: /Users/calebjasik/Git/jasik-2.0/dist/
13:25:17 [build] adapter: @astrojs/netlify
13:25:17 [build] Collecting build info...
13:25:17 [build] ✓ Completed in 886ms.
13:25:17 [build] Building server entrypoints...
13:25:17 [WARN] [vite] [vite:react-babel] We recommend switching to `@vitejs/plugin-react-oxc` for improved performance. More information at https://vite.dev/rolldown
[sentry-vite-plugin] Info: Sending telemetry data on issues and performance to Sentry. To disable telemetry, set `options.telemetry` to `false`.
`transformWithEsbuild` is deprecated and will be removed in the future. Please migrate to `transformWithOxc`.
13:25:18 [WARN] [vite] [vite:css][lightningcss] Unknown at rule: @theme
1  |  
2  |        @layer theme, base, components, utilities;
   |       ^
3  |        @import "tailwindcss/theme" layer(theme);
4  |        @import "tailwindcss/utilities" layer(utilities);
13:25:18 [WARN] [vite] [vite:css][lightningcss] Unknown at rule: @theme

13:25:18 [WARN] [vite] [vite:css][lightningcss] Unknown at rule: @tailwind
1  |  
2  |        @layer theme, base, components, utilities;
   |          ^
3  |        @import "tailwindcss/theme" layer(theme);
4  |        @import "tailwindcss/utilities" layer(utilities);
13:25:19 [ERROR] [vite] ✗ Build failed in 1.49s
[builtin:transform] Error: Tsconfig not found astro/tsconfigs/strict

  Location:
    /Users/calebjasik/Git/jasik-2.0/node_modules/.pnpm/rolldown@1.0.0-beta.38/node_modules/rolldown/dist/shared/src-CWMXqoXE.mjs:2135:18
  Stack trace:
    at normalizeErrors (file:///Users/calebjasik/Git/jasik-2.0/node_modules/.pnpm/rolldown@1.0.0-beta.38/node_modules/rolldown/dist/shared/src-CWMXqoXE.mjs:2135:18)
    at transformToRollupOutput (file:///Users/calebjasik/Git/jasik-2.0/node_modules/.pnpm/rolldown@1.0.0-beta.38/node_modules/rolldown/dist/shared/src-CWMXqoXE.mjs:2873:2)
    at async buildEnvironment (file:///Users/calebjasik/Git/jasik-2.0/node_modules/.pnpm/rolldown-vite@7.1.9_@types+node@22.16.4_esbuild@0.25.9_jiti@2.4.2_yaml@2.8.0/node_modules/rolldown-vite/dist/node/chunks/dep-HNGlhFR2.js:33174:64)
    at async ssrBuild (file:///Users/calebjasik/Git/jasik-2.0/node_modules/.pnpm/astro@5.12.2_@netlify+blobs@10.0.10_@types+node@22.16.4_idb-keyval@6.2.1_jiti@2.4.2_rol_a2c32aaf5948e7ea7c89cb0b2a0e6bb7/node_modules/astro/dist/core/build/static-build.js:181:10)
    at async AstroBuilder.build (file:///Users/calebjasik/Git/jasik-2.0/node_modules/.pnpm/astro@5.12.2_@netlify+blobs@10.0.10_@types+node@22.16.4_idb-keyval@6.2.1_jiti@2.4.2_rol_a2c32aaf5948e7ea7c89cb0b2a0e6bb7/node_modules/astro/dist/core/build/index.js:149:48)
 ELIFECYCLE  Command failed with exit code 1.
```

--- a/crates/rolldown_plugin_transform/src/utils.rs
+++ b/crates/rolldown_plugin_transform/src/utils.rs
@@ -119,7 +119,7 @@ impl TransformPlugin {
     }
 
     if source_type.is_typescript() {
-      let path = Path::new(id).parent().and_then(find_tsconfig_json_for_file);
+      let path = Path::new(cwd).join(id).parent().and_then(find_tsconfig_json_for_file);
       let tsconfig = path.map(|path| ctx.resolver().resolve_tsconfig(&path)).transpose()?;
 
       if let Some(tsconfig) = tsconfig {

